minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-runtime/src/main/java/org/apache/nifi/minifi/MiNiFi.java

```diff
@@ -216,12 +221,43 @@ public void run() {
      * @param args things which are ignored
      */
     public static void main(String[] args) {
-        logger.info("Launching MiNiFi...");
+        LOGGER.info("Launching MiNiFi...");
         try {
-            NiFiProperties niFiProperties = MultiSourceMinifiProperties.getInstance();
-            new MiNiFi(niFiProperties);
+            NiFiProperties properties = getValidatedMiNifiProperties(args);
+            new MiNiFi(properties);
         } catch (final Throwable t) {
-            logger.error("Failure to launch MiNiFi due to " + t, t);
+            LOGGER.error("Failure to launch MiNiFi due to " + t, t);
+        }
+    }
+
+    protected static NiFiProperties getValidatedMiNifiProperties(String[] args) {
+        NiFiProperties properties = initializeProperties(args, createBootstrapClassLoader());
+        properties.validate();
+        return properties;
+    }
+
+    private static NiFiProperties initializeProperties(final String[] args, final ClassLoader boostrapLoader) {
+        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+        String key = getFormattedKey(args);
+
+        Thread.currentThread().setContextClassLoader(boostrapLoader);
+
+        try {
+            Class<?> propsLoaderClass = Class.forName("org.apache.nifi.minifi.properties.MiNiFiPropertiesLoader", true, boostrapLoader);
+            Method withKeyMethod = propsLoaderClass.getMethod("withKey", String.class);
+            Object loaderInstance = withKeyMethod.invoke(null, key);
```

Comment:
Is this approach necessary for MiNiFI? The reflection construct is required for NiFi, but it would be helpful to know if it is required for MiNiFi or whether this can be simplified.

Reply:
Thanks for confirming, that's good to know.